server/filestore.go

```diff
@@ -5213,6 +5221,10 @@ func (mb *msgBlock) readPerSubjectInfo(hasLock bool) error {
 		}
 	}()
 
+	// Make sure we run the cache expire timer.
+	mb.llts = time.Now().UnixNano()
```

Comment:
Btw, should this be protected by the lock, which is acquired only later on in this function when invoked with !hasLock?

Reply:
True wil fix..